source/elements/oneDAL/source/algorithms/decomposition/pca.rst

```diff
@@ -1,3 +1,447 @@
+         | ``desc& set_component_count(std::int64_t)``
+
+      Invariants
+         | :expr:`component_count >= 0`
+
+
+   .. member:: bool set_is_deterministic = true
+
+      Specifies whether the algorithm applies `Sign-flip technique`_ or uses
+      deterministic eigenvalues solver. If it is `true`, directions of the
+      eigenvectors must be deterministic.
+
+      Getter & Setter
+         | ``bool get_is_deterministic() const``
+         | ``desc& set_is_deterministic(bool)``
+
+
+Model
+-----
+::
+
+   class model {
+   public:
+      model();
+
+      const table& get_eigenvectors() const;
+      int64_t get_component_count() const;
+   };
+
+.. class:: model
+
+   .. function:: model()
+
+      Creates model with default attribute values.
+
+
+   .. member:: table eigenvectors = table()
+
+      :math:`r \times p` table with the eigenvectors. Each row contains one
+      eigenvector.
+
+      Getter
+         | ``const table& get_eigenvectors() const``
+
+
+   .. member:: std::int64_t component_count = 0
+
+      The number of components :math:`r` in the trained model.
+
+      Getter
+         | ``std::int64_t get_component_count() const``
+
+      Invariants
+         | :expr:`component_count == centroids.row_count`
+
+
+
+Training :expr:`onedal::train(...)`
+-----------------------------------
+Input
+~~~~~
+::
+
+   class train_input {
+   public:
+      train_input();
+      train_input(const table& data);
+
+      const table& get_data() const;
+
+      train_input& set_data(const table&);
+   };
+
+.. class:: train_input
+
+   .. function:: train_input()
+
+      Creates an input for the training operation with the default attribute
+      values.
+
+
+   .. function:: train_input(const table& data)
+
+      Creates an input for the training operation with the given :expr:`data`.
+
+
+   .. member:: table data = table()
+
+      :math:`n \times p` table with the data to be clustered, where each row
+      stores one sample.
+
+      Getter & Setter
+         | ``const table& get_data() const``
+         | ``train_input& set_data(const table&)``
+
+
+Result
+~~~~~~
+::
+
+   class train_result {
+   public:
+      train_result();
+
+      const model& get_model() const;
+      const table& get_means() const;
+      const table& get_variances() const;
+      const table& get_eigenvalues() const;
+   };
+
+.. class:: train_result
+
+   .. function:: train_result()
+
+      Creates a result of the training operation with the default attribute
+      values.
+
+
+   .. member:: pca::model model = pca::model()
+
+      The trained K-means model.
+
+      Getter
+         | ``const model& get_model() const``
+
+
+   .. member:: table means = table()
+
+      :math:`1 \times r` table that contains mean value for each feature.
+
+      Getter
+         | ``const table& get_means() const``
+
+
+   .. member:: table variances = table()
+
+      :math:`1 \times r` table that contains variance for each feature.
```

Comment:
those r and each may not be clean - it means for each i-th feature, i=1,...,r/for the first r features?